src/GenerateRoutes.zig

```diff
@@ -0,0 +1,272 @@
+
+    var walker = try views_dir.walk(self.allocator);
+    defer walker.deinit();
+
+    while (try walker.next()) |entry| {
+        if (entry.kind != .file) continue;
+
+        const extension = std.fs.path.extension(entry.path);
+        const basename = std.fs.path.basename(entry.path);
+
+        if (std.mem.eql(u8, basename, "routes.zig")) continue;
+        if (std.mem.eql(u8, basename, "zmpl.manifest.zig")) continue;
+        if (std.mem.startsWith(u8, basename, ".")) continue;
+        if (!std.mem.eql(u8, extension, ".zig")) continue;
+
+        const routes = try self.generateRoute(views_dir, entry.path);
+
+        for (routes.static) |route| {
+            try self.static_routes.append(route);
+        }
+
+        for (routes.dynamic) |route| {
+            try self.dynamic_routes.append(route);
+        }
+    }
+
+    std.sort.pdq(Function, self.static_routes.items, {}, Function.lessThanFn);
+    std.sort.pdq(Function, self.dynamic_routes.items, {}, Function.lessThanFn);
+
+    try writer.writeAll("pub const routes = struct {\n");
+    try writer.writeAll("    pub const static = .{\n");
+
+    for (self.static_routes.items) |static_route| {
+        try self.writeRoute(writer, static_route);
+    }
+
+    try writer.writeAll("    };\n\n");
+    try writer.writeAll("    pub const dynamic = .{\n");
+
+    for (self.dynamic_routes.items) |dynamic_route| {
+        try self.writeRoute(writer, dynamic_route);
+        const name = try dynamic_route.fullName(self.allocator);
+        defer self.allocator.free(name);
+        std.debug.print("[jetzig] Imported route: {s}\n", .{name});
+    }
+
+    try writer.writeAll("    };\n");
+    try writer.writeAll("};");
+}
+
+fn writeRoute(self: *Self, writer: std.ArrayList(u8).Writer, route: Function) !void {
+    const full_name = try route.fullName(self.allocator);
+    defer self.allocator.free(full_name);
+
+    const uri_path = try route.uriPath(self.allocator);
+    defer self.allocator.free(uri_path);
+
+    const output_template =
+        \\        .{{
+        \\            .name = "{s}",
+        \\            .action = "{s}",
+        \\            .uri_path = "{s}",
+        \\            .template = "{s}",
+        \\            .function = @import("{s}").{s},
+        \\        }},
+        \\
+    ;
+
+    const output = try std.fmt.allocPrint(self.allocator, output_template, .{
+        full_name,
+        route.name,
+        uri_path,
+        full_name,
+        route.path,
+        route.name,
+    });
+
+    defer self.allocator.free(output);
+    try writer.writeAll(output);
+}
+
+const RouteSet = struct {
+    dynamic: []Function,
+    static: []Function,
+};
+
+fn generateRoute(self: *Self, views_dir: std.fs.Dir, path: []const u8) !RouteSet {
+    // REVIEW: Choose a sensible upper limit or allow user to take their own risks here ?
+    const stat = try views_dir.statFile(path);
+    const source_buf = try views_dir.readFileAlloc(self.allocator, path, stat.size);
```

Comment:
The zig ast is limited to 4GB, so is a fair assumption that the max size we should allow is 4GB

Reply:
Do you think fair to let Zig handle that ? i.e. leave it as is, let Zig fail if AST reaches >4G ?

I think I prefer this otherwise we could potentially have a false-positive error if Zig changes this limit in future.

Reply:
I don't mind, that's the reason I put all this as comments, Don't think is a blocker